### PR TITLE
[#4279] feat(client-python): Add the `getFileLocation` interface code skeleton in Python Client

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Added an interface code skeleton in Python Client for obtaining the file location so that the client can report some necessary information for the server to audit and simplify some check logics in Python GVFS later. Depend on #4320.

### Why are the changes needed?

Fix: #4279 

### How was this patch tested?

Add UTs and ITs.

### CODE REVIEW

**Comment by xloya** [comment]
@jerryshao @noidname01 This PR is ready for review, please take a look of this when you have time, thanks.

File: clients/client-python/gravitino/audit/fileset_data_operation.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# specific language governing permissions and limitations
+# under the License.
+from enum import Enum
+
+
+class FilesetDataOperation(Enum):
+    """An enum class containing fileset data operations that supported."""
+
+    CREATE = "CREATE"
+    """ This data operation means that create a new file.
```
Comment (jerryshao) [inline clients/client-python/gravitino/audit/fileset_data_operation.py:24]: I think the comment can be simplified as "Creates a new file", "Options a file", that should be enough.

Comment (xloya) [reply clients/client-python/gravitino/audit/fileset_data_operation.py]: Done

File: clients/client-python/gravitino/audit/caller_context.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# specific language governing permissions and limitations
+# under the License.
+import threading
+from typing import Dict
+
+caller_context_holder = threading.local()
+
+
+class CallerContext:
+    """A class defining the caller context for auditing coarse granularity operations."""
```
Comment (jerryshao) [inline clients/client-python/gravitino/audit/caller_context.py:24]: "auditing coarse-grained operations."

**Comment by jerryshao** [comment]
Generally LGTM, just one comment.

File: clients/client-python/gravitino/audit/caller_context.py (hunk 2 above)
Comment (xloya) [reply clients/client-python/gravitino/audit/caller_context.py]: Done

**Review by jerryshao — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]

**Pull request merged**